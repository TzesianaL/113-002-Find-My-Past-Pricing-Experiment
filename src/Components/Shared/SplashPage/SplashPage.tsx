import React, { FC } from 'react'
import StoreBanner from 'Components/Shared/StoreBanner/StoreBanner'
import Title from 'Components/Shared/Title/Title'
import { SplashItem } from 'types'
import Button from 'Components/Shared/Button/Button'
import mainImage from 'images/main_image.png'
import awards from 'images/splash-bl.png'
import fouthPanel from 'images/splash-br.png'
import mainImagePhoneVersion from 'images/main_image_phone_version.png'
import secondImagePhoneVersion from 'images/second_image_phone_version.png'
import main_faster from 'images/main_faster.png'
import main_faster_mobile from 'images/main_faster_mobile.png'
import {
  SplashPageStyle,
  SplashWrapper,
  ContentGrid,
  AwardsImage,
  DesktopImageContainer,
  MobileImageContainer,
} from './SplashPageStyle'
import { useSelector } from 'react-redux'
import { getSelectedCondition } from 'redux/reducers/dataStore/dataStore.selectors'

export interface SplashPageProps {
  /**
   * storeLogo has the path for the store logo
   */
  storeLogo: string
  /**
   * Contains the name of the store each time.
   */
  storeName: string
  /**
   * pageTitle displays the Main page's title.
   */
  pageTitle: string
  /**
   * items contains title and text of each block.
   */
  items: SplashItem[]
  /**
   * buttonText has the text to be shown on the button
   */
  buttonText: string
  /**
   * function defines what action will the button does.
   */
  onClick: () => void
}

/**
 * SplashScreen component is used to group together all splashscreen components.
 * SplashScreenStyle component is used to style it.
 */
const SplashPage: FC<SplashPageProps> = ({
  storeLogo,
  storeName,
  pageTitle,
  items,
  buttonText,
  onClick,
}) => {
  const selectedCondition = useSelector(getSelectedCondition)

  const mainImageSrc =
    selectedCondition === 1 || selectedCondition === 2 ? mainImage : main_faster
  const mainImagePhoneSrc =
    selectedCondition === 1 || selectedCondition === 2
      ? mainImagePhoneVersion
      : main_faster_mobile
  return (
    <SplashWrapper>
      <StoreBanner name={storeName} logoSrc={storeLogo} />
      <ContentGrid>
        <div style={{ gridColumnStart: 1, gridColumnEnd: 3 }}>
          <Title
            style={{
              fontSize: '2rem',
              paddingTop: '1em',
              paddingBottom: '1em',
            }}
          >
            Smith’s Broadband Offers You
          </Title>
          <DesktopImageContainer>
            <img src={mainImageSrc} alt="main-image" width="100%" />
          </DesktopImageContainer>
          <MobileImageContainer>
            <img src={mainImagePhoneSrc} alt="main-image" width="400px" />
            <img
              src={secondImagePhoneVersion}
              alt="second-image"
              width="400px"
            />
          </MobileImageContainer>
        </div>
        <section
          style={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            margin: '0 1rem',
          }}
        >
          <Title
            style={{
              fontSize: '1.75rem',
              textAlign: 'left',
              width: '100%',
              paddingLeft: '16px',
              paddingRight: '16px',
              boxSizing: 'border-box',
              marginBottom: '0',
              padding: '0.5rem 0 1rem 0',
            }}
          >
            Award-winning broadband and reviews
          </Title>
          <img src={awards} alt="awards" width="100%" />
        </section>
        <section
          style={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'left',
            textAlign: 'left',
            margin: '0 1rem',
          }}
        >
          <Title
            style={{
              fontSize: '1.75rem',
              textAlign: 'left',
              width: '100%',
              padding: '0.5rem 0 1rem 0',
            }}
          >
            Switching to Smith’s is easy
          </Title>
          <img
            src={fouthPanel}
            alt="fourthPanel"
            width="100%"
            style={{ paddingTop: '9px' }}
          />
        </section>
      </ContentGrid>
      <br />
      <br />
      <Button onClick={onClick}>{buttonText}</Button>
      <br />
      <br />
    </SplashWrapper>
  )
}

export default SplashPage
