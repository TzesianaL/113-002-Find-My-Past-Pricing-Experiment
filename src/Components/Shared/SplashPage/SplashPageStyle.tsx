import styled from 'styled-components'
import { TitleStyle } from '../Title/TitleStyle'
/**
 * Styled component is used to decorate SplashScreen component.
 * Props are passed from ThemeProvider component.
 */
export const SplashPageStyle = styled.div`
  display: grid;
  justify-content: center;
  grid-template-columns: 40% 40%;
  // grid-template-rows: 40% 40%;
  row-gap: 3%;
  column-gap: 3%;
  padding-bottom: 3%;
  max-width: 1440px;
  @media screen and (max-width: 675px) {
    display: flex;
    flex-direction: column;
  }
`
export const SplashWrapper = styled.div`
  display: grid;
  margin-bottom: 10em;
  justify-items: center;
  background: #e5e5e5;

  ${TitleStyle} {
    padding-top: 2em;
    padding-bottom: 2em;
  }
`

export const ContentGrid = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  align-items: start;
  width: 100%;

  @media screen and (max-width: 768px) {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
`

export const AwardsImage = styled.img`
  width: 90%;

  @media screen and (max-width: 768px) {
    width: 100%;
  }
`

export const DesktopImageContainer = styled.div`
  display: block;
  margin: 17px;

  @media screen and (max-width: 768px) {
    display: none;
  }
`

export const MobileImageContainer = styled.div`
  display: none;
  flex-direction: column;
  align-items: center;
  width: 100%;

  @media screen and (max-width: 768px) {
    display: flex;
  }
`
