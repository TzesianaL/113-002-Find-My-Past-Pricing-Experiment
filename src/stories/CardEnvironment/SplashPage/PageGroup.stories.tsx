import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import { ThemeProvider } from 'styled-components'
import SplashPage, {
  SplashPageProps,
} from 'Components/Shared/SplashPage/SplashPage'
import { handleConsoleLogClick } from 'functions/functions'
import { testingItems } from './__fixtures__/splashpage.fixtures'

export default {
  title: 'Shared/SplashPage/SplashPage',
  component: SplashPage,
} as Meta

interface StyledSplashPageProps extends SplashPageProps {
  backgroundColor: string
  textColor: string
  fontSize: number
  fontFamily: string
  bannerColor: string
  bannerBackgroundColor: string
  titleColor: string
  titleFontSize: number
  imageSize: number
  buttonWidth: string
  borderRadius: string
  activeBackgroundButtonColor: string
}

const Template: Story<StyledSplashPageProps> = ({
  backgroundColor,
  textColor,
  fontSize,
  fontFamily,
  bannerColor,
  bannerBackgroundColor,
  titleColor,
  titleFontSize,
  imageSize,
  buttonWidth,
  borderRadius,
  activeBackgroundButtonColor,
  ...args
}) => (
  <ThemeProvider
    theme={{
      titleFontSize: titleFontSize || 3,
      fontFamily: `${fontFamily || 'Arial'}`,
      titleColor: `${titleColor || 'grey'}`,
      splashScreenText: {
        backgroundColor: `${backgroundColor || 'lightgrey'}`,
        color: `${textColor || 'black'}`,
        fontSize: fontSize || 1,
        fontFamily: `${fontFamily || 'Arial'}`,
      },
      splashImage: {
        imageSize: imageSize || 100,
      },

      banner: {
        color: bannerColor,
        backgroundColor: bannerBackgroundColor,
        fontSize: 1.2,
      },
      button: {
        color: `${textColor || 'black'}`,
        backgroundColor: `${backgroundColor || 'lightgrey'}`,
        width: `${buttonWidth || '25%'}`,
        borderRadius: `${borderRadius || '2rem'}`,
        activeBackgroundColor: `${activeBackgroundButtonColor || 'grey'}`,
      },
    }}
  >
    <SplashPage {...args} />
  </ThemeProvider>
)

export const Basic = Template.bind({})

Basic.args = {
  storeName: 'StoreName',
  pageTitle: 'PageTitle',
  items: testingItems,
  backgroundColor: '#D8D8D8',
  textColor: '#505050',
  fontSize: 1,
  fontFamily: 'Arial',
  bannerColor: '#E6E6FA',
  bannerBackgroundColor: '#8B008B',
  titleColor: 'red',
  titleFontSize: 2,
  imageSize: 46,
  buttonText: 'Button',
  onClick: handleConsoleLogClick,
}

Basic.argTypes = {
  backgroundColor: {
    control: 'color',
  },
  textColor: {
    control: 'color',
  },
  bannerColor: {
    control: 'color',
  },
  bannerBackgroundColor: {
    control: 'color',
  },
  titleColor: {
    control: 'color',
  },
}
