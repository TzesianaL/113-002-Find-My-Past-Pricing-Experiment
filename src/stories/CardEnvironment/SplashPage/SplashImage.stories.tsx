import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import SplashImg, {
  SplashImgProps,
} from 'Components/Shared/SplashPage/SplashImage/SplashImage'
import img from '../../images/logo.png'

export default {
  title: 'Environments/Card Environment/SplashScreen/SplashImage',
  component: SplashImg,
} as Meta

const Template: Story<SplashImgProps> = (args) => <SplashImg {...args} />

export const Basic = Template.bind({})

Basic.args = {
  imageSrc: img,
}
