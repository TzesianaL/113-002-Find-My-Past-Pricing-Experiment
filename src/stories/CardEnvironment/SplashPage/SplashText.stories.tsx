import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import { ThemeProvider } from 'styled-components'
import SplashText, {
  SplashTextProps,
} from 'Components/Shared/SplashPage/SplashText/SplashText'

export default {
  title: 'Environments/Card Environment/SplashScreen/SplashText',
  component: SplashText,
} as Meta

interface StyledSplashTextProps extends SplashTextProps {
  backgroundColor: string
  textColor: string
  fontSize: number
  fontFamily: string
}

const Template: Story<StyledSplashTextProps> = ({
  backgroundColor,
  textColor,
  fontSize,
  fontFamily,
  ...args
}) => (
  <ThemeProvider
    theme={{
      splashScreenText: {
        backgroundColor: `${backgroundColor || 'lightgrey'}`,
        color: `${textColor || 'black'}`,
        fontSize: fontSize || 1,
        fontFamily: `${fontFamily || 'Arial'}`,
      },
    }}
  >
    <SplashText {...args} />
  </ThemeProvider>
)

export const Basic = Template.bind({})

Basic.args = {
  title: 'Title',
  text: 'p1 <b>test test test test </b>p2 <br/>p3',
  backgroundColor: 'lightgrey',
  textColor: 'grey',
  fontSize: 1,
  fontFamily: 'Arial',
}
Basic.argTypes = {
  backgroundColor: {
    control: 'color',
  },
  textColor: {
    control: 'color',
  },
}
