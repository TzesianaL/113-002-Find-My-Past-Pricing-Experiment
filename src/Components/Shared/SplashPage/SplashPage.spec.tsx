import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import { testingItems } from 'stories/CardEnvironment/SplashPage/__fixtures__/splashpage.fixtures'
import SplashPage from './SplashPage'

describe('<SplashPage/>', () => {
  it('should render correctly SplashPage', () => {
    const tree = renderer
      .create(
        <SplashPage
          storeLogo="tets storelogo"
          storeName="test store name"
          pageTitle="test page title"
          items={testingItems}
          buttonText="test button text"
          onClick={() => {}}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
