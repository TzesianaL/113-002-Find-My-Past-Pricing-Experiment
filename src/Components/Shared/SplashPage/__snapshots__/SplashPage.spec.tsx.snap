// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashPage/> should render correctly SplashPage 1`] = `
.c1 {
  background-color: #6300AA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fill-available;
  width: -moz-available;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white !important;
  padding: 0.6rem 0 0.6rem 0;
}

.c1 p {
  width: 100%;
  font-size: rem;
}

.c1 img {
  padding: 0 1em;
}

.c3 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  font-family: BoldVf !important;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c7 {
  color: #ffffff;
  background-color: #003D59;
  cursor: pointer;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0px;
  border-radius: 6px;
  padding: 0.7em 1em;
}

.c7:active {
  background-color: #336D89;
}

.c6 {
  background-color: #6300AA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: white !important;
  font-size: rem;
  padding: 16px 1em 24px 1em;
}

.c5 {
  display: grid;
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100%;
}

.c5 img {
  width: 100%;
}

.c4 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  grid-template-columns: 40% 40%;
  row-gap: 3%;
  -webkit-column-gap: 3%;
  column-gap: 3%;
  padding-bottom: 3%;
  max-width: 1440px;
}

.c0 {
  display: grid;
  margin-bottom: 10em;
  justify-items: center;
  background: #333333;
}

.c0 .c2 {
  padding-top: 2em;
  padding-bottom: 2em;
}

@media screen and (max-width:675px) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="logo"
      height="44"
      src="tets storelogo"
    />
    <p>
      test store name
    </p>
  </div>
  <div
    className="c2 c3"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "test page title",
        }
      }
    />
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <img
        alt="splashimg"
        src="cardHeader.png"
      />
    </div>
    <div
      className="c6"
    >
      <div
        className="c2 c3"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Vodafone",
            }
          }
        />
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<b>Minutes: 300</b><br/> Texts: 100 Data: 20GB",
          }
        }
      />
    </div>
    <div
      className="c6"
    >
      <div
        className="c2 c3"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Vodafone",
            }
          }
        />
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<b>Minutes: 300</b><br/> Texts: 100 Data: 20GB",
          }
        }
      />
    </div>
    <div
      className="c6"
    >
      <div
        className="c2 c3"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Vodafone",
            }
          }
        />
      </div>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<b>Minutes: 300</b><br/> Texts: 100 Data: 20GB",
          }
        }
      />
    </div>
  </div>
  <button
    className="c7"
    onClick={[Function]}
  >
    test button text
  </button>
  <br />
  <br />
</div>
`;
